cordova-plugin-outline/apple/vpn/Shadowsocks.m

```diff
@@ -178,42 +172,17 @@ - (void) checkServerConnectivity {
     self.startCompletion(noError);
     return;
   }
-  __block BOOL isRemoteUdpForwardingEnabled = false;
-  __block BOOL serverCredentialsAreValid = false;
-  __block BOOL isServerReachable = false;
-
-  // Enter the group once for each check
-  dispatch_group_enter(self.dispatchGroup);
-  dispatch_group_enter(self.dispatchGroup);
-  dispatch_group_enter(self.dispatchGroup);
-
-  [self.ssConnectivity isUdpForwardingEnabled:^(BOOL enabled) {
-    isRemoteUdpForwardingEnabled = enabled;
-    dispatch_group_leave(self.dispatchGroup);
-  }];
-  [self.ssConnectivity isReachable:self.config[@"host"]
-                              port:[self.config[@"port"] intValue]
-                        completion:^(BOOL isReachable) {
-                          isServerReachable = isReachable;
-                          dispatch_group_leave(self.dispatchGroup);
-                        }];
-  [self.ssConnectivity checkServerCredentials:^(BOOL valid) {
-    serverCredentialsAreValid = valid;
-    dispatch_group_leave(self.dispatchGroup);
-  }];
-
-  dispatch_group_notify(self.dispatchGroup, self.dispatchQueue, ^{
-    DDLogInfo(@"Server connectivity checks done");
-    if (isRemoteUdpForwardingEnabled) {
-      self.startCompletion(noError);
-    } else if (serverCredentialsAreValid) {
-      self.startCompletion(udpRelayNotEnabled);
-    } else if (isServerReachable) {
-      self.startCompletion(invalidServerCredentials);
-    } else {
-      self.startCompletion(serverUnreachable);
-    }
-  });
+  long errorCode;
+  NSError *error;
+  ShadowsocksCheckConnectivity(self.config[@"host"], [self.config[@"port"] intValue],
```

Comment:
Apple seems to recommend using the return value in this situation, rather than dispatching on whether `error` is `nil`.

Reply:
The function's return type is void. When a Go function returns an error as a single parameter, gomobile creates "idiomatic bindings", which in Obj-C means setting an error pointer (an exception is thrown in Java).

Reply:
I completely missed the `BOOL` return in the bindings, nor coud I find any docs. Done now.